internal/bminventory/inventory.go

```diff
@@ -4929,6 +4922,27 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 		return nil, common.NewApiError(http.StatusBadRequest, err)
 	}
 
+	var cluster *common.Cluster
+	clusterId := infraEnv.ClusterID
+	if clusterId != "" {
+		cluster, err = b.GetClusterInternal(ctx, installer.V2GetClusterParams{ClusterID: clusterId})
+		if err != nil {
+			// We don't want to fail here if cluster is not found. It's not responsability of this place
+			// to verify that, so if there is a real issue with non-existing cluster it will be detected
+			// and raised by someone else.
+			cluster = nil
+		}
+	}
+	if err = validateArchitectureAndVersion(b.versionsHandler, cluster, infraEnv.CPUArchitecture, infraEnv.OpenshiftVersion); err != nil {
+		return nil, err
+	}
+	if cluster != nil {
+		if err = featuresupport.ValidateIncompatibleFeatures(log, infraEnv.CPUArchitecture, *cluster, &infraEnv.InfraEnv, nil); err != nil {
+			b.log.Error(err)
+			return nil, common.NewApiError(http.StatusBadRequest, err)
+		}
+	}
+
```

Comment:
This code was moved from line 4952 because it was misplaced.
The changes were saved to the DB and only than validated the CPU architecture and the OCP Version, meaning that infraenv can be updated with invalid arguments and fail after the update